compiler/src/main/java/com/squareup/hephaestus/compiler/InterfaceMerger.kt

```diff
@@ -107,4 +124,12 @@ internal class InterfaceMerger(
     supertypes += contributedClasses
     super.addSyntheticSupertypes(thisDescriptor, supertypes)
   }
+
+  private fun List<KotlinType>.getAllSuperTypes(): List<FqName> =
+    generateSequence(this) { kotlinTypes ->
+      if (kotlinTypes.isEmpty()) null else kotlinTypes.flatMap { it.supertypes() }
```

Comment:
Before you merge, check if you should return emptyList() here instead of null.

Reply:
Yes, that's important and correct. Without returning `null` `generateSequence` would end in an endless loop and never finish.